pytest_inmanta/plugin.py

```diff
@@ -615,6 +656,30 @@ def get_context_for(
             )
         return self.results[resources[0]]
 
+    def get_resource(
+        self, resource_type: str, **filter_args: object
+    ) -> typing.Optional[Resource]:
+        """
+        Get a resource of the given type and given filter on the results attributes from a Result object.
+        If multiple resource match, the first one is returned. If none match, None is returned.
+
+        :param resource_type: The exact type used in the model (no super types)
+        """
+        return get_resource(self.results.keys(), resource_type, **filter_args)
+
+
+class DeployResult(Result):
+    """
+    Here for backwards compatibility reasons.
+    """
+
+
+@dataclass
+class ResultCollection:
```

Comment:
```suggestion
class DryrunResultCollection:
```
Because it's not just any collection of results.

Reply:
It is also not just any collection of dryrun results, since we actually deploy there. Maybe `DeployResultCollecton` might be better, because technically a dryrun is just a simulated deploy

Reply:
Sounds good to me.